design/one-pager-template-stacks.md

```diff
@@ -0,0 +1,399 @@
+
+### No Custom Controller Needed
+
+Existing Stacks must specify a `Deployment`. These are typically written in Go and generally require understanding some combination of client-go, controller-runtime, crossplane-runtime, and Kubebuilder.
+
+For some applications, the intended work of the controller could be simple:
+
+* For every new resource, create a set of Kubernetes resources
+* Use the spec fields of that resource to affect the Kubernetes resources
+* Upon changes to these spec fields, update the Kubernetes resources
+
+This is where the value of Template Stacks starts to materialize.
+
+```yaml
+kind: Stack
+metadata:
+  name: RedisStack
+spec:
+  templates:
+    redis.example.org/v1:
+      deployment: |
+          …
+          containers:
+          - name: redis-controller
+            image: example/redis-controller:{{.spec.redisVersion}}
+```
+
+*In this example, templates are nested within a map keyed by the GVR and a template grouping name. This will be explained in the [One Pager for Template Stack Fields](one-pager-template-stack-fields).*
+
+### Composition
+
+#### Vertical Composition
+
+Crossplane Stacks manage multiple CRDs. Template Stacks should be no different.
+
+Template definitions will need to express the template to render for each CRD that is managed. The GVR can be used as a map key to denote the CRD template being described.
+
+Each template needed to render a GVR should be provided (demonstrated below as a map), additionally a name that represents that intent of that template should distinguish each template (also demonstrated as a map). This name can then be reused to reference a template (see [Horizontal Composition](#Horizontal-Composition)).
+
+```yaml
+kind: Stack
+metadata:
+  name: CachingWebServiceStack
+spec:
+  templates:
+    redis.example.org/v1:
+      deployment: |
+        kind: Deployment
+        …
+          containers:
+          - name: redis-controller
+            image: example/redis-controller:{{.spec.redisVersion}}
+    cachingwebservice.example.org/v1:
+      cache: |
+        apiVersion: example.org/v1
+        kind: Redis
+        …
+        spec:
+          redisVersion: "5"
+      web: |
+        kind: Deployment
+        …
+          containers:
+          - name: nginx-controller
+            image: example/nginx-controller:{{.spec.nginxVersion}}
+--
+kind: CachingWebService
+spec:
+  nginxVersion: "1.17.4"
+```
+
+In the example above, distinct templates are defined for the Redis CRD and the CachingWebService. CachingWebService includes two templates, one of which relies on the other.
+
+The `nginxVersion` is defined in the `spec` of the `CachingWebService` instance, while `redisVersion` has been declared in the `cache` template for `cachingwebservice.example.org/v1`. The template variable has been passed into a descendent Stack.
```

Comment:
What is a `descendent Stack`?

Reply:
Beats me..

I'm replacing that last sentence with:

> In each case, a template variable has been passed into a dependent template. One was user supplied, the other was template supplied.

Thanks for pointing this out.

Related to this, I have started using "dependent resource" to describe resources the templated sub-resources of a Template Stack.  Look for this in further updates and let me know how it fits?

Reply:
In general I am not a huge fan of the term `dependent` because it is so similar to `dependency`; I like to use some version of `consume` or `consumer` when possible so that the terms are More Different.

I'll keep an eye out for the terminology changes though!